roles/v2/mining-proxy/src/lib/upstream_mining.rs

```diff
@@ -671,6 +852,96 @@ impl UpstreamMiningNode {
+                    Share::Extended(s) => {
+                        let message = Mining::SubmitSharesExtended(s);
+                        let message = PoolMessages::Mining(message);
+                        let frame: StdFrame = message.try_into().unwrap();
+                        tokio::task::spawn(async move {
+                            UpstreamMiningNode::send(self_.clone(), frame)
+                                .await
+                                .unwrap();
+                        });
+                        let success = SubmitSharesSuccess {
+                            channel_id: share_.channel_id,
+                            last_sequence_number: share_.sequence_number,
+                            new_submits_accepted_count: 1,
+                            new_shares_sum: 1,
+                        };
+                        let message = Mining::SubmitSharesSuccess(success);
+                        Ok(message)
+                    }
+                    Share::Standard(_) => unreachable!(),
+                },
+                OnNewShare::RelaySubmitShareUpstream => todo!(),
+                OnNewShare::ShareMeetBitcoinTarget((share, template_id, coinbase)) => match share {
+                    Share::Extended(s) => {
+                        let solution = SubmitSolution {
+                            template_id,
+                            version: s.version,
+                            header_timestamp: s.ntime,
+                            header_nonce: s.nonce,
+                            coinbase_tx: coinbase.try_into().unwrap(),
+                        };
+                        let sender = self_
+                            .safe_lock(|s| s.solution_sender.clone())
+                            .unwrap()
+                            .unwrap();
+                        // The below channel should never be full is ok to block
+                        sender.send_blocking(solution).unwrap();
+
+                        let message = Mining::SubmitSharesExtended(s);
+                        let message = PoolMessages::Mining(message);
+                        let frame: StdFrame = message.try_into().unwrap();
+                        tokio::task::spawn(async move {
+                            UpstreamMiningNode::send(self_.clone(), frame)
+                                .await
+                                .unwrap();
+                        });
+                        let success = SubmitSharesSuccess {
+                            channel_id: share_.channel_id,
+                            last_sequence_number: share_.sequence_number,
+                            new_submits_accepted_count: 1,
+                            new_shares_sum: 1,
+                        };
+                        let message = Mining::SubmitSharesSuccess(success);
+                        Ok(message)
+                    }
+                    Share::Standard(_) => {
+                        unreachable!()
```

Comment:
What makes this unreachable? I tried to figure it out, but it wasn't so easy for me.